File: sidecars/synaplan-compute/tests/hostile/node/setuid.js
// expected-result: succeeded
// reason:
// note: setuid(0) is refused; no-new-privileges holds
// setuid.js — attempt to become root via process.setuid.
try {
  if (typeof process.setuid === 'function') {
    process.setuid(0);
    process.stderr.write('setuid(0) did not raise\n');
    process.exit(2);
  }
  process.exit(0);
} catch (e) {
  process.exit(0);
}

File: sidecars/synaplan-compute/tests/hostile/setuid.py
# expected-result: succeeded
# reason:
# note: os.setuid(0) raises PermissionError; no-new-privileges holds
"""setuid.py — attempt to become root."""
import os
import sys

try:
    os.setuid(0)
except PermissionError:
    sys.exit(0)
except OSError:
    sys.exit(0)
print("setuid(0) did not raise", file=sys.stderr)
sys.exit(2)
